crates/ln-dlc-node/src/node/dlc_channel.rs

```diff
@@ -158,6 +158,73 @@ impl<S: TenTenOneStorage + 'static, N: LnDlcStorage + Sync + Send + 'static> Nod
         Ok(())
     }
 
+    pub async fn close_dlc_channel(&self, channel_id: DlcChannelId, force: bool) -> Result<()> {
+        let channel_id_hex = hex::encode(channel_id);
+        tracing::info!(channel_id = channel_id_hex, "Closing DLC channel");
+
+        let channel = self
+            .get_dlc_channel(|channel| channel.channel_id == channel_id)?
+            .context("DLC channel to close not found")?;
+
+        if force {
+            // TODO(bonomat): implement force closure
+        } else {
+            self.propose_dlc_channel_collaborative_close(channel)
+                .await?
+        }
+
+        Ok(())
+    }
+
+    /// Collaboratively close a DLC channel on-chain collaboratively if there is no open position
+    async fn propose_dlc_channel_collaborative_close(&self, channel: SignedChannel) -> Result<()> {
+        let channel_id_hex = hex::encode(channel.channel_id);
+
+        tracing::info!(
+            channel_id = %channel_id_hex,
+            "Closing DLC channel collaboratively"
+        );
+        self.is_safe_to_close_dlc_channel_collaboratively(&channel)?;
+        let counterparty = channel.counter_party;
+
+        match channel.state {
+            SignedChannelState::Settled { .. } | SignedChannelState::RenewFinalized { .. } => {
+                spawn_blocking({
+                    let dlc_manager = self.dlc_manager.clone();
+                    let dlc_message_handler = self.dlc_message_handler.clone();
+                    let peer_manager = self.peer_manager.clone();
+                    move || {
+                        let settle_offer = dlc_manager
+                            .offer_collaborative_close(
+                                &channel.channel_id,
+                                channel.counter_params.collateral,
+                            )
+                            .context(
+                                "Could not propose to collaboratively close the dlc channel.",
+                            )?;
+
+                        send_dlc_message(
+                            &dlc_message_handler,
+                            &peer_manager,
+                            counterparty,
+                            Message::Channel(ChannelMessage::CollaborativeCloseOffer(settle_offer)),
+                        );
+
+                        anyhow::Ok(())
+                    }
+                })
+                .await??;
+            }
+            _ => {
+                tracing::error!(  state = %channel.state, "Can't collaboratively close a channel with an open position.");
+                bail!("Can't collaboratively close a channel with an open position");
```

Comment:
🔧 Why not?

Also, it's weird to log and return an error for the same error.

Reply:
> Also, it's weird to log and return an error for the same error.

I started doing this because I stopped trusting where errors are being logged (or even ignored) and with this it's easier to find them.

> 🔧 Why not?

Mhm, good question. We had this in the past, so I kept it. Happy to change it.

Reply:
Thinking about it more. It's easier to tell the user to close the position first because otherwise we will need to calculate the PnL here and think about how to handle the otherside of the trade.

Reply:
> I started doing this because I stopped trusting where errors are being logged (or even ignored) and with this it's easier to find them.

Hmmm, I'm not happy with this because we will end up spamming the logs with redundant messages, which makes debugging harder. We just need to fix the spots where errors are being swallowed. I don't think there can be many of those.

Reply:
> Thinking about it more. It's easier to tell the user to close the position first because otherwise we will need to calculate the PnL here and think about how to handle the otherside of the trade.

Fair. It's not strictly necessary and it's nice to keep things simple.